docs/antrea-network-policy.md

```diff
@@ -1037,13 +887,82 @@ spec:
         name: AllowToCoreDNS
 ```
 
-The above example allows all Pods from Namespace "default" to connect to all "core-dns"
+The above example allows all Pods from Namespace "default" to connect to all "kube-dns"
 Pods from Namespace "kube-system" on TCP port 53.
 
-## FQDN based filtering
+### Matching self Namespace for appliedTos
 
-Antrea-native policy accepts a `fqdn` field to select Fully Qualified Domain Names (FQDNs),
-specified either by exact name or wildcard expressions, when defining `egress` rules.
+The `namespaces` field allows users to perform advanced matching on Namespace objects that cannot 
+be done via label selectors. Currently, the `namespaces` field has only one matching strategy, `Self`. 
+If set to `Self`, for each Pod affected by `appliedTo` of the policy/rule, this field selects the same
+Namespace that the Pod is in. It enables policy writers to create per-Namespace rules within a
+single policy.
+
+Consider a minimalistic cluster, where there are only three Namespaces labeled ns=x, ns=y and ns=z.
+Inside each of these Namespaces, there are three Pods labeled app=a, app=b and app=c.
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: allow-self-ns
+spec:
+  priority: 1
+  tier: platform
+  appliedTo:
+    - namespaceSelector: {}
+  ingress:
+    - action: Allow
+      from:
+        - namespaces:
+            match: Self
+    - action: Deny
+  egress:
+    - action: Allow
+      to:
+        - namespaces:
+            match: Self
+    - action: Deny
+```
+
+The policy above ensures that x/a, x/b and x/c can communicate with each other, but nothing else
+(unless there are higher precedenced policies which says otherwise). Same for Namespaces y and z.
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: deny-self-ns-a-to-b
+spec:
+  priority: 1
+  tier: securityops
+  appliedTo:
+    - namespaceSelector: {}
+      podSelector:
+        matchLabels:
+          app: b
+  ingress:
+    - action: Deny
+      from:
+        - namespaces:
+            match: Self
+          podSelector:
+            matchLabels:
+              app: a
+```
+
+The `deny-self-ns-a-to-b` policy ensures that traffic from x/a -> x/b, y/a -> y/b and z/a -> z/b are
+denied. It can be used in conjunction with the `allow-self-ns` policy. If both policies are applied,
```

Comment:
either "traffic from x/a to x/b, y/a to y/b and z/a to z/b is denied"
or
"traffic x/a -> x/b, y/a -> y/b and z/a -> z/b is denied"

Reply:
Done